### DISCUSSION TITLE
Renovate doesn't update package.json

### DISCUSSION THREAD

**Discussion by rgrzesk** [renovatebot/renovate#22822] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Hi!

I have been using Renovate for some time for npm based project. It worked perfectly until this week it started creating PRs with `package-lock.json` updates only. It doesn't update `package.json` at all. Where to look for a problem?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment, marked as answer]
What you're describing is a feature, not a bug. Look up `rangeStrategy` in the docs to understand more about how to change it if you desire.

**Reply by rgrzesk** [reply]
Thank you @rarkins ! I thought that it might be an issue since it was working before correctly.
I can see that most of the cases are usually mentioning how to not update `package-lock.json`. What should I set to update both?

**Reply by rarkins** [reply]
I'm going to assume you have cases where your `package.json` has a dependency set to `^1.0.0`, your lock file has `1.1.0`, and Renovate detects that the highest matching version is `1.2.0`. It's then giving you a "lockfile-only update" bumping your locked version from `1.1.0` to `1.2.0`.

So the first question is, why do you think the `package.json` needs to change too? The proposed update is perfect valid.

Please read https://docs.renovatebot.com/dependency-pinning/ as a background

**Comment by rgrzesk** [comment]
@rarkins that was the first stuff that happened. Then I closed the PRs and there is another issue - the packages are removed completely from the lock file. It seems it is another case. So: there is a PR done by renovate with a title of upgrading some library and in the commits there is a change with removing that library. Would that be a bug or rather some misconfiguration?

**Reply by rarkins** [reply]
Hard to tell without a reproduction to demonstrate it. If you can do that, please create a new discussion as it sounds unrelated to this one
